service/worker/deletenamespace/deleteexecutions/workflow.go

```diff
@@ -110,6 +110,12 @@ func DeleteExecutionsWorkflow(ctx workflow.Context, params DeleteExecutionsParam
 	runningDeleteExecutionsSelector := workflow.NewSelector(ctx)
 	var lastDeleteExecutionsActivityErr error
 
+	// Two activities DeleteExecutionsActivity and GetNextPageTokenActivity are executed here essentially in reverse order
```

Comment:
Now I think this optimization is not worth it. The approach that migration wf took to have one visibility call (by local activity) and pass the result to activity to process them is simpler and does not seem to have any perf issue.